### PR TITLE
sys-apps: Add a port of util-linux

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR adds a stripped down port of `util-linux` and performs some updates.

The following ports have been added:
- `util-linux`, albeit in a stripped down form, missing various utilities one might expect from this package, like `swapon` or `agetty`. All libraries provided by this package, like `libblkid`, `libfdisk` and `libmount` are available.

The following ports received updates:
- `ncurses`, we now build a seperate `tinfo` library as `util-linux` seems to like that, I'm not quite sure why. Furthermore, we now build the wide character versions of the libraries, making compatibility symlinks to the non-wide version as is common on Linux systems,
- `bash`, added a direct dependency on ncurses and force a rebuild to link against the new ncurses libraries,
- `nano`, override configure on some decisions and force a rebuild to link against the new ncurses libraries,
- `vim`, disable the use of the X windowing system even harder and force a rebuild to link against the new ncurses libraries,
- `less`, force a rebuild to link against the new ncurses libraries.

Blocked on managarm/mlibc#214
Blocked on managarm/mlibc#222

### CODE REVIEW

File: bootstrap.d/sys-libs.yml [hunk 1]
```diff
@@ -64,6 +64,7 @@ packages:
       version: '6.1'
     tools_required:
       - system-gcc
+    revision: 3
```
Comment (avdgrinten) [inline bootstrap.d/sys-libs.yml:67]: Why do we bump from 1 to 3?

File: patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+--- a/libmount/src/context_umount.c
++++ b/libmount/src/context_umount.c
+@@ -291,7 +291,9 @@ static int lookup_umount_fs_by_statfs(struct libmnt_context *cxt, const char *tg
+ 		DBG(CXT, ul_debugobj(cxt, "  trying fstatfs()"));
+ 
+ 		/* O_PATH avoids triggering automount points. */
+-		fd = open(tgt, O_PATH);
++		/* MANAGARM: We don't support O_PATH (yet) so disable it. */
++		fd = open(tgt, 0);
++		//fd = open(tgt, O_PATH);
```
Comment (avdgrinten) [inline patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch:124]: What happens if we enable this?

File: patches/util-linux/0002-Revamp-the-configure-script-to-allow-for-more-fine-g.patch [hunk 3]
```diff
@@ -0,0 +1,488 @@
+From 629cb91f214d29f55744515577aa5ca898094434 Mon Sep 17 00:00:00 2001
+From: Dennis Bonke <admin@dennisbonke.com>
+Date: Sat, 23 Jan 2021 00:06:46 +0100
+Subject: [PATCH 2/2] Revamp the configure script to allow for more
+ fine-grained control over the programs build. Make configure think that
+ managarm is Linux.
```
Comment (avdgrinten) [inline patches/util-linux/0002-Revamp-the-configure-script-to-allow-for-more-fine-g.patch:6]: The commit message of this patch does not have a proper caption.

File: patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+ # include <libmount.h>	/* keep it optional for non-linux systems */
+diff --git a/include/namespace.h b/include/namespace.h
+index 2d0a56e02..75d0f439d 100644
+--- a/include/namespace.h
++++ b/include/namespace.h
+@@ -7,6 +7,7 @@
+  */
+ #ifndef UTIL_LINUX_NAMESPACE_H
+ # define UTIL_LINUX_NAMESPACE_H
++#ifndef __managarm__
```
Comment (avdgrinten) [inline patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch:40]: What do we patch out here? (And in other places?)

File: patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
++++ b/term-utils/agetty.c
+@@ -88,6 +88,14 @@
+ #  ifndef DEFAULT_STERM
+ #    define DEFAULT_STERM  "vt102"
+ #  endif
++#elif defined __managarm__
++#  define USE_SYSLOG
++#  ifndef DEFAULT_VCTERM
++#    define DEFAULT_VCTERM "managarm"
++#  endif
```
Comment (avdgrinten) [inline patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch:166]: Is this a terminal name? If yes, I don't think we should define a "managarm" terminal type.

**Review by avdgrinten — COMMENTED** [review]
I have a few comments about the patch to the C / header files. The rest looks good.

File: patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch (hunk 4 above)
Comment (Dennisbonke) [reply patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch:40]: Several linux only features like namespaces or the Linux specific cpu interface

File: patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch (hunk 5 above)
Comment (Dennisbonke) [reply patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch:166]: I think it indeed it. I can change it to match Linux there, that may be more appropriate.

Comment (avdgrinten) [reply patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch:166]: Agreed.

**Review by avdgrinten — APPROVED** [review]
LGTM now.

**Pull request merged**